libnmstate/nm/ipv4.py

```diff
@@ -119,3 +126,95 @@ def get_ip_profile(active_connection):
     if remote_conn:
         return remote_conn.get_setting_ip4_config()
     return None
+
+
+def get_route_info():
+    ret = []
+    client = nmclient.client()
+    for active_connection in client.get_active_connections():
+        if active_connection is None:
+            continue
+        ip_cfg = active_connection.get_ip4_config()
+        if not ip_cfg:
+            continue
+
+        devs = active_connection.get_devices()
+        iface_name = None
+        # If we have master device, we use it, else we use first device.
+        if hasattr(active_connection, 'master'):
+            iface_name = active_connection.master.get_iface()
+        else:
+            devs = active_connection.get_devices()
+            if devs:
+                iface_name = devs[0].get_iface()
+
+        if not iface_name:
+            logging.warning(
+                'Got connection {} has not interface name'.format(
+                    active_connection.get_id()))
+            continue
+
+        skip_networks = list('{}/{}'.format(addr.get_address(),
+                                            addr.get_prefix())
+                             for addr in ip_cfg.get_addresses())
+        skip_networks.append(iplib.IPV4_LINK_LOCAL_NETWORK)
+
+        static_routes = []
+        ip_profile = get_ip_profile(active_connection)
+        if ip_profile:
+            for route in ip_profile.props.routes:
+                static_routes.append(
+                    '{ip}/{prefix}'.format(
+                        ip=route.get_dest(), prefix=route.get_prefix()))
+
+        table_id = _get_route_table_id(active_connection, ip_cfg)
+        for route in ip_cfg.get_routes():
+            dst = '{ip}/{prefix}'.format(
+                ip=route.get_dest(), prefix=route.get_prefix())
+            if _should_skip(dst, skip_networks):
+                continue
+            route_origin = 'static'
+            if active_connection.get_dhcp4_config() and \
+               dst not in static_routes:
+                route_origin = 'dhcp'
+
+            next_hop = route.get_next_hop()
+            if not next_hop:
+                next_hop = ''
+            metric = int(route.get_metric())
+            route_entry = {
+                'status': 'up',
```

Comment:
We started using constants for field names, therefore it would be great to use them here as well. See the OVS code changes for example.

Reply:
OK. Will do.

Reply:
Done.